### ISSUE TITLE
DISALLOW_TYPE_COERCION and ALLOW_TYPE_COERCION values

### ISSUE THREAD

**Issue by richiware** [OpenDDS/OpenDDS#4709]
I think `DISALLOW_TYPE_COERCION` and `ALLOW_TYPE_COERCION` are not using the correct values. [Here](https://github.com/OpenDDS/OpenDDS/blob/master/dds/DdsDcpsCore.idl#L298) `DISALLOW_TYPE_COERCION` is set to `1` and `ALLOW_TYPE_COERCION` is set to `2`.

These literals are defined in standard DDS X-Types 1.3 as
```
@bit_bound(16)
enum TypeConsistencyKind {
  DISALLOW_TYPE_COERCION,
  ALLOW_TYPE_COERCION
};
```
and also in the same standard in section **7.3.1.2.1.5 Enumerated Literal Values** it says:
```
In such cases, as in C and C++ enumerations, implicit values are assigned in a
progression starting from the most-recently specified value (or an implicit value of zero for the
first literal, if there is no previous specified value) and adding one with each successive literal.
```

Shouldn't `DISALLOW_TYPE_COERCION` have the value 0 and `ALLOW_TYPE_COERCION` have the value 1?

Thanks in advance.

**Comment by iguessthislldo** [comment]
I don't think it matters for OpenDDS, because as far as I know we don't support most of type consistency at the moment, but yes, it should be fixed.
